egui/src/widgets/text_edit.rs

```diff
@@ -182,9 +186,71 @@ impl<'t> TextEdit<'t> {
             enabled: true,
             desired_width: None,
             desired_height_rows: 4,
+            tab_as_spaces: false,
+            tab_moves_focus: true,
         }
     }
 
+    /// Registers if this widget will insert spaces instead of tab char
+    ///
+    /// ```rust, ignore
+    /// ui.add(egui::TextEdit::multiline(&mut self.multiline_text_input)
+    ///     .tab_as_spaces(true));
+    /// ```
+    pub fn tab_as_spaces(mut self, b: bool) -> Self {
+        self.tab_as_spaces = b;
+        self
+    }
+
+    /// When this is true, then pass focus to the next
+    /// widget.
+    ///
+    /// When this is false, then insert identation based on the value of
+    /// `tab_as_spaces` property.
+    ///
+    /// ```rust, ignore
+    /// ui.add(egui::TextEdit::multiline(&mut self.multiline_text_input)
+    ///     .tab_moves_focus(true));
+    /// ```
+    pub fn tab_moves_focus(mut self, b: bool) -> Self {
+        self.tab_moves_focus = b;
+        self
+    }
+
+    /// Build a `TextEdit` focused on code editing.
+    /// By default it comes with:
+    /// - monospaced font
+    /// - focus lock
+    /// - tab as spaces
+    ///
+    /// Shortcut for:
+    /// ```rust, ignore
+    /// egui::TextEdit::multiline(code_snippet)
+    ///     .text_style(TextStyle::Monospace)
+    ///     .tab_as_spaces(true)
+    ///     .tab_moves_focus(false);
+    /// ```
+    pub fn code_editor(self) -> Self {
+        self.text_style(TextStyle::Monospace)
+            .tab_as_spaces(true)
+            .tab_moves_focus(false)
+    }
+
+    /// Build a `TextEdit` focused on code editing with configurable `Tab` management.
+    ///
+    /// Shortcut for:
+    /// ```rust, ignore
+    /// egui::TextEdit::multiline(code_snippet)
+    ///     .code_editor()
+    ///     .tab_as_spaces(tab_as_spaces)
+    ///     .tab_moves_focus(tab_moves_focus);
+    /// ```
+    pub fn code_editor_with_config(self, tab_as_spaces: bool, tab_moves_focus: bool) -> Self {
```

Comment:
Same comment here about multiple boolean arguments. If you want to set multiple values with a single call, it is better to make a config struct and pass that in (`struct CodeConfig { tabs_as_spaces: bool, tab_width: usize, … } `). This has the benefit of being more explicit and more extensible.

Reply:
I crated a `CodingConfig` struct for this purpose.
https://github.com/DrOptix/egui/blob/e14c5491b61574647ed837130e0a61f4c1ce98b4/egui/src/widgets/text_edit.rs#L107-L112